src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/StringExtensions.cs

```diff
@@ -33,6 +33,7 @@ internal static partial class StringExtensions
 #else
 		private static readonly Lazy<Regex> _newLineRegex = new Lazy<Regex>(() => new Regex(@"^", RegexOptions.Multiline));
 #endif
+		public static Uri TrimEndUriSlash(this Uri uri) => uri.ToString().EndsWith('/') ? new Uri(uri.ToString().TrimEnd(new[] { '/' })) : uri;
```

Comment:
I suppose that it is more resource-efficient to call `uri.AbsolutePath.EndsWith('/')` then materializing the whole `Uri` into a `string`

I propose such changes

```suggestion
		public static Uri TrimEndUriSlash(this Uri uri) => uri.AbsolutePath.EndsWith('/') ? new Uri(uri.ToString()[..^1]) : uri;
```

Reply:
Nice @xperiandri, I commi your suggestion, thnak you!